File: 2110.number-of-smooth-descent-periods-of-a-stock/solution.go
func getDescentPeriods(prices []int) int64 {
    previousPrice := 0
    previousStart := -1
    res := int64(0)

    for i, p := range prices {
        if p != previousPrice - 1 {
            previousStart = i
        }
        res += int64(i - previousStart + 1)
        previousPrice = p
    }

    return res
}

File: 2110.number-of-smooth-descent-periods-of-a-stock/solution.kt
class Solution {
    fun getDescentPeriods(prices: IntArray): Long {
        var previousPrice = -1
        var previousStart = -1
        var res = 0L

        prices.forEachIndexed { i, price ->
            if (price != previousPrice - 1) {
                previousStart = i
            }
            res += i - previousStart + 1
            previousPrice = price
        }

        return res
    }
}